assets/src/components/ui/icon-menu-item/entity-action-menu-item.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External imports
+ */
+import { Component } from 'react';
+
+/**
+ * EntityActionMenuItem
+ * just a wrapper for an IconButton and additional component
+ * (most likely a modal component) so that it can be added to a menu
+ */
+class EntityActionMenuItem extends Component {
+	render() {
+		return this.props.children;
+	}
+}
```

Comment:
hmm... this would be better as a functional component:
```jsx
const EntityActionMenuItem = ( { children } ) => children;
```

What is the use-case for this though (i.e. how would it be implemented?  It's not clear from the description or the code why it's needed.  It "appears" it's just used for conditional checks?

I'm probably going to be implementing the slot/fill pattern for the action menu as well (which will work a bit better for extensibility than the current filter hook method).

Reply:
ya I severely want to limit what can be injected into the entity action menus.
basically we only ever want icon buttons, but even amongst the WP core components there are a few that are viable (like the drop down menu), so I unfortunately can't limit it to just `IconButton`s.
This is essentially just a wrapper that can be used, which I realize doesn't actually prevent someone from injecting anything they want.
I've considered using this as a parent class that others could inherit, but that's somewhat frowned upon in React in favour of composition, which doesn't really apply in this use case.

I'm open to other ideas for improving this

Reply:
Yup I grok the desire to limit what can be added to this area.  I think we could explore some further restriction ideas when I get to the slot/fill extension task.